cmd/brew-test-bot.rb

```diff
@@ -775,7 +775,15 @@ def bottle_reinstall_formula(formula, new_formula)
       test "brew", "bottle", *bottle_merge_args
       test "brew", "uninstall", "--force", formula.name
 
-      FileUtils.ln bottle_filename, HOMEBREW_CACHE/bottle_filename, force: true
+      bottle_json = JSON.parse(File.read(bottle_json_filename))
+
+      root_url = bottle_json.dig(formula.name, "bottle", "root_url")
+      filename = bottle_json.dig(formula.name, "bottle", "tags").values.first["filename"]
+      url_sha256 = Digest::SHA256.hexdigest("#{root_url}/#{filename}")
+
+      FileUtils.ln bottle_filename, HOMEBREW_CACHE/"downloads/#{url_sha256}--#{filename}", force: true
+      FileUtils.ln_s "downloads/#{url_sha256}--#{filename}", HOMEBREW_CACHE/bottle_filename, force: true
```

Comment:
Could this maybe get refactored into some method (doesn't have to be public) in Homebrew/brew that test-bot can call instead? Don't really like the logic being duplicated here.

Reply:
Any suggestion on where to put it? I thought about it but I can't think of a good place where this logic should go. Even in the download strategies, these steps don't happen in a single cohesive action, and `FileUtils.ln` is specific to `test-bot`.